articles/extensions/custom-extensions.md

```diff
@@ -8,35 +8,109 @@ To begin creating an your custom extension, please feel free to fork/clone any o
 
-At this point, you have two options in addition to completing the installation of the extension. You may:
+## Configuring the Installed Extension
 
-- complete the installation at a later date. If you click on the "X" at the top right to exit the window, Auth0 will add your extension to the "All Extensions" tab. At a later date, you may choose this to complete the installation;
-- delete the extension completely by clicking on the "Delete" button.
+Under the _Installed Extensions_ tab you will find your newly-installed extension listed.
 
-## Configuring the Installed Extension
+![](/media/articles/extensions/custom/installed-extensions.png)
+
+Click on the link to obtain more information about the app. You will be prompted to authorize the extension's access to your Auth0 account.
+
+![](/media/articles/extensions/custom/authorize-app.png)
+
+After you've authorized the extension's access to your account, you will be presented with a [JSON Web Token](/tokens/).
+
+![](/media/articles/extensions/custom/json-web-token.png)
+
+## Extension Lifecycle
+
+Let's have a look at what happens behind the scenes when installing and uninstalling custom extensions.
 
-Once you have installed your extension, you...
+When the user clicks on _Install_, a Client is created for the extension with the scopes defined on the `webtask.json` and access is granted to APIv2 Resource Server.
 
-Under the "Installed Extensions" tab you will find your newly-installed extension listed. Click on the link to obtain more information about the app. You will be prompted to authorize the extension's access to your Auth0 account.
+For this `webtask.json`:
 
-After you've authorized the extension's access to your account, you will be presented with a [JSON Web Token](/tokens/) that is similar to the following:
+```json
+{
+  "name": "my-extension";
+  "auth0": {
+    "createClient": true,
+    "scopes": "create:rules"
+  }
+}
+```
+
+The following Client would be created:
+
+```javascript
+Clients.create({
+  name: 'my-extension'
+}).then(function (client) {
+  return Grants.create({
+    audience:  'https://jcenturion.auth0.com/api/v2/',
+    client_id: client.client_id,
+    scope: "create:rules"
+  }).then(function () {
+    return addSecrets(wt, client, wtUrl);
+  });
+});
+```
+
+The installation dialog will warn the user that the extension will have access to certain scopes. In this case: `create:rules`.
+
+![](/media/articles/extensions/custom/scopes-warning.png)
+
+The webtask will be created with the secrets `AUTH0_CLIENT_ID` and `AUTH0_CLIENT_SECRET`.
 
+After the webtask is created, `/.webtask/on-install` (`POST /onInstallUrl`) is called sending a [JWT](/jwt) for validating that Auth0-manage is the one calling it.
+
+> The expected success status is `204`. Keep in mind that if the hooks fail, then the install (or uninstall) will fail as well.
+
+Install and uninstall URLs are configurable through `webtask.json`.
```

Comment:
`onInstallPath` and `onUninstallPath` are mandatory if you want `auth0-dashboard` to call them.